src/org/labkey/signaldata/SignalDataController.java

```diff
@@ -123,7 +123,7 @@ public void setTest(boolean test)
 
 
     @RequiresPermission(ReadPermission.class)
-    public class getSignalDataResourceAction extends ApiAction<SignalDataResourceForm>
+    public class getSignalDataResourceAction extends ReadOnlyApiAction<SignalDataResourceForm>
     {
```

Comment:
This is a MutatingApiAction - line 155 does data.save(), which does an insert/update to LK db.

Reply:
thanks for catching this, how about the permission? Does that need to be updated?